develop/blockchains/deployment/generate-chain-specs.md

```diff
@@ -1,4 +1,136 @@
+
+- **Client Specification** - contains information the _node_ uses to communicate with network participants and send data to telemetry endpoints. Many of these chain specification settings can be overridden by command-line options when starting a node or can be changed after the blockchain has started
+
+- **Initial Genesis State** -  agreed upon by all nodes in the network. It must be set when the blockchain is first started and cannot be changed after that without starting a whole new blockchain
+
+## Node Settings Customization
+
+For the node, the chain specification controls information such as:
+
+- The bootnodes the node will communicate with
+- The server endpoints for the node to send telemetry data to
+- The human and machine-readable names for the network the node will connect to
+
+The chain specification can be customized to include additional information. For example, you can configure the node to connect to specific blocks at specific heights to prevent long-range attacks when syncing a new node from genesis.
+
+Note that you can customize node settings after genesis. However, nodes only add peers that use the same [`protocolId`](https://paritytech.github.io/polkadot-sdk/master/polkadot_parachain_lib/chain_spec/trait.ChainSpec.html#tymethod.protocol_id){target=_blank}.
+
+## Genesis Configuration Customization
+
+All nodes in the network must agree on the genesis state before they can agree on any subsequent blocks. The information configured in the genesis portion of a chain specification is used to create a genesis block. When you start the first node, it takes effect and cannot be overridden with command-line options. However, you can configure some information in the genesis section of a chain specification. For example, you can customize it to include information such as:
+
+- Initial account balances
+- Accounts that are initially part of a governance council
+- The account that controls the `sudo` key
+- Any other genesis state for a pallet
+
+Nodes also require the compiled Wasm to execute the runtime logic on the chain, so the initial runtime must also be supplied in the chain specification.
+
+## Declaring storage items for a runtime
+
+A runtime usually requires some storage items to be configured at genesis. This includes the initial state for pallets, for example, how much balance specific accounts have, or which account will have sudo permissions.
+
+These storage values are configured in the genesis portion of the chain specification. You can create a [patch](https://paritytech.github.io/polkadot-sdk/master/sc_chain_spec/index.html#chain-spec-formats){target=_blank} file for [`chain-spec-builder`](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/reference_docs/chain_spec_genesis/index.html){target=_blank} to ingest or use `chain_spec.rs`, and the pallet genesis state can be configured in the source code.
```

Comment:
hi @michalkucharczyk ! Thank you for the feedback

What would be the example precisely? I'm having trouble finding it

Reply:
If I understand correctly, you are looking for something like this:
https://github.com/paritytech/polkadot-sdk/blob/a4791617241732f4fecafe21e12c16b7fe42f73e/cumulus/polkadot-parachain/src/chain_spec/asset_hubs.rs#L136-L151


But instead of calling:
```
.with_genesis_config_preset_name("genesis")
```
you want to call:
```
.with_genesis_config_patch(json!({...})
```

like in this code:
https://github.com/paritytech/polkadot-sdk/blob/a4791617241732f4fecafe21e12c16b7fe42f73e/substrate/client/chain-spec/src/chain_spec.rs#L964-L988

I think we don't have a pure json patches in our codebase - all chain specs are now based on embedded presets.